tests/ztest/error_hook/src/main.c

```diff
@@ -51,16 +51,16 @@ static void trigger_fault_illeagl_instuction(void)
 
 static void trigger_fault_access(void)
 {
-#if defined(CONFIG_CPU_ARCEM)
+#if defined(CONFIG_CPU_ARCEM) || defined(CONFIG_CPU_CORTEX_M)
```

Comment:
Hi, I have run a trial on this, even I keep the both original one and new one (with/without || defined(CONFIG_CPU_CORTEX_M), then this error_hook test case can still pass on the qemu platform. So I guess this change is necessary in the physical cortex-M board, is my understanding right?  Thank you!

Reply:
Yes! So let me explain:
In ARM Cortex M with CONFIG_USERSPACE=y, read access to address 0x0 is generally allowed by the kernel, if the address is mapped, ie. it is part of the image. So, the explicit event of de-referencing NULL is not generally causing a CPU fault: so the assumption in the test is wrong.

The reasons that the test is observed to be passing are the following:
1. 0x0 address is unmapped. This is the case of the QEMU (mps2_an385) platform. In this case we do get an MPU fault (fatal error) because 0x0 is not given un-privileged (user) access.
2. 0x0 address is mapped, but the compiler detects at build time the access to NULL and adds an undefined instruction. This is the case observed in nRF52 builds.

We have recently introduced the NULL-pointer detection feature via DWT, so access to NULL triggers an additional DebugMonitor fault. So if we are in the case (1), we are going to get both MPU and DebugMonitor fault, and the test will fail because it is not designed to handle multiple fault exceptions. 

You do not see this in QEMU, because QEMU does not support the null-pointer exception via DebugMonitor.

Reply:
Thanks so much your clear explanation, now I understand and I really appreciate your help on adapting this to error hook test case.

Reply:
Could you add the explanation in the comment below? This will allow the next reader to understand why Cortex-M needs this. Otherwise, LGTM.

Reply:
done @dcpleung pls, take a look. I only added the part that explains why NULL pointer dereferencing is not good for cortex-m here :)